### PR TITLE
createdBefore & after filters in branches page

### PR DESCRIPTION
**Pull request by montedonio-pinterest**
- createdBefore & after filters in branches page
- cleanup code & remove warns from console
    - add key prop to .map functions in render method
    - component prop expected string but got num
    - component prop expected value but got null at page rendering

### CODE REVIEW

File: webapp/src/main/resources/public/js/stores/PaginatorStore.js [hunk 1]
```diff
@@ -34,7 +34,9 @@ class PaginatorStore {
     }
 
     changeCurrentPageNumber(currentPageNumber) {
-        this.currentPageNumber = currentPageNumber;
+        this.currentPageNumber = currentPageNumber
```
Comment (garionpin) [inline webapp/src/main/resources/public/js/stores/PaginatorStore.js:37]: Do you have any context why this change is needed (were we not displaying the page number in certain scenarios?)?

Comment (garionpin) [reply webapp/src/main/resources/public/js/stores/PaginatorStore.js:37]: Thanks for clarifying!

File: webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js [hunk 2]
```diff
@@ -45,6 +46,20 @@ class BranchesHistoryStore {
         this.openBranchStatistic = openBranchStatistic;
     }
 
+    changeCreatedBefore(createdBefore) {
+        this.createdBefore = createdBefore
```
Comment (aurambaj) [inline webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js:50]: all other methods are simple setters so this should stay a simple setter as well. What call sites needs to be updated to mimic that behavior if we're keeping a normal setter?

File: webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js [hunk 3]
```diff
@@ -119,6 +135,12 @@ class BranchesHistoryStore {
         BranchesSearchParamsActions.changeEmpty(empty === "true");
         BranchesSearchParamsActions.changeNotEmpty(notEmpty === "true");
         BranchesSearchParamsActions.changeOnlyMyBranches(onlyMyBranches === "true");
+        if (createdBefore !== null) {
+            BranchesSearchParamsActions.changeCreatedBefore(moment(createdBefore))
```
Comment (aurambaj) [inline webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js:139]: looks like that's where we want to apply those conversion. similar to what we do for `selectedBranchTextUnitIds`

File: webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js (hunk 2 above)
Comment (montedonio-pinterest) [reply webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js:50]: Was able to simplify these setters by handling the value as strings and letting the Datetime component convert to the moment object and handle the parsing (which it already did by default 😅 )

File: webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js (hunk 3 above)
Comment (montedonio-pinterest) [reply webapp/src/main/resources/public/js/stores/branches/BranchesHistoryStore.js:139]: Handling as strings now, don't even need to convert!

**Review by aurambaj — APPROVED** [review]
Looks good!
